docs/content/platform/new_blocks.md

```diff
@@ -286,32 +286,155 @@ Finally you will need to add the provider to the `CredentialsType` enum in [`fro
 
-```python title="blocks/google/_auth.py"
+```python title="backend/blocks/google/_auth.py"
 --8<-- "autogpt_platform/backend/backend/blocks/google/_auth.py:GoogleOAuthIsConfigured"
 ```
 
 You can also see that `GOOGLE_OAUTH_IS_CONFIGURED` is used to disable the blocks that require OAuth if the oauth is not configured. This is in the `__init__` method of each block. This is because there is no api key fallback for google blocks so we need to make sure that the oauth is configured before we allow the user to use the blocks.
 
+### Webhook-triggered Blocks
+
+Webhook-triggered blocks allow your agent to respond to external events in real-time.
+These blocks are triggered by incoming webhooks from third-party services
+rather than being executed manually.
+
+Creating and running a webhook-triggered block involves three main components:
+
+- The block itself, which specifies:
+    - Inputs for the user to select a resource and events to subscribe to
+    - A `credentials` input with the scopes needed to manage webhooks
+    - Logic to turn the webhook payload into outputs for the webhook block
+- The `WebhooksManager` for the corresponding webhook service provider, which handles:
+    - (De)registering webhooks with the provider
+    - Parsing and validating incoming webhook payloads
+- The credentials system for the corresponding service provider, which may include an `OAuthHandler`
+
+There is more going on under the hood, e.g. to store and retrieve webhooks and their
+links to nodes, but to add a webhook-triggered block you shouldn't need to make changes
+to those parts of the system.
+
+#### Creating a Webhook-triggered Block
+
+To create a webhook-triggered block, follow these additional steps on top of the basic block creation process:
+
+1. **Define `webhook_config`** in your block's `__init__` method.
+
+    <details>
+    <summary>Example: <code>GitHubPullRequestTriggerBlock</code></summary>
+
+    ```python title="backend/blocks/github/triggers.py"
+    --8<-- "autogpt_platform/backend/backend/blocks/github/triggers.py:example-webhook_config"
+    ```
+    </details>
+
+    <details>
+    <summary><code>BlockWebhookConfig</code> definition</summary>
+
+    ```python title="backend/data/block.py"
+    --8<-- "autogpt_platform/backend/backend/data/block.py:BlockWebhookConfig"
+    ```
+    </details>
+
+2. **Define event filter input** in your block's Input schema.
+
+    <details>
+    <summary>Example: <code>GitHubPullRequestTriggerBlock</code></summary>
+
+    ```python title="backend/blocks/github/triggers.py"
+    --8<-- "autogpt_platform/backend/backend/blocks/github/triggers.py:example-event-filter"
+    ```
+    </details>
+
+    - The name of the input field (`events` in this case) must match `webhook_config.event_filter_input`.
+    - The event filter itself must be a Pydantic model with only boolean fields.
+
+3. **Include payload field** in your block's Input schema.
```

Comment:
Can this be a pydantic model?

Reply:
In theory, sure. I'm not sure the existing implementation of webhook ingress endpoint + executor supports it though.